src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -4113,6 +4113,16 @@ void Unit::RemoveArenaAuras()
     }
 }
 
+void Unit::RemoveAurasOnEvade()
+{
+    if (IsPet() || IsGuardian()) // if pet or guardian shouldn't lose any aura
```

Comment:
do we have a check for such creatures? (personally i never saw more than IsPet() and IsGuardian() in any check for player owned creatures)